docs/core/diagnostics/observability-with-otel.md

```diff
@@ -0,0 +1,426 @@
+current_connections{endpoint="127.0.0.1:7275"} 1 1686894204856
+current_connections{endpoint="[::1]:7275"} 0 1686894204856
+current_connections{endpoint="[::1]:5212"} 1 1686894204856
+...
+```
+
+The metrics output is presented in prometheus format which is human readable, but better understood by prometheus which we will get to in the next stage.
+
+If you look at the console for the server, you'll see the output from the console trace provider. This should show two activities, one from our custom ActivitySource, and the other from ASP.NET Core:
+
+``` shell
+Activity.TraceId:            2e00dd5e258d33fe691b965607b91d18
+Activity.SpanId:             3b7a891f55b97f1a
+Activity.TraceFlags:         Recorded
+Activity.ParentSpanId:       645071fd0011faac
+Activity.ActivitySourceName: OtPrGrYa.Sample
+Activity.DisplayName:        GreeterActivity
+Activity.Kind:               Internal
+Activity.StartTime:          2023-06-16T04:50:26.7675469Z
+Activity.Duration:           00:00:00.0023974
+Activity.Tags:
+    greeting: Hello World!
+Resource associated with Activity:
+    service.name: OTel-Prometheus-Graphana-Yaeger
+    service.instance.id: e1afb619-bc32-48d8-b71f-ee196dc2a76a
+    telemetry.sdk.name: opentelemetry
+    telemetry.sdk.language: dotnet
+    telemetry.sdk.version: 1.5.0
+
+Activity.TraceId:            2e00dd5e258d33fe691b965607b91d18
+Activity.SpanId:             645071fd0011faac
+Activity.TraceFlags:         Recorded
+Activity.ActivitySourceName: Microsoft.AspNetCore
+Activity.DisplayName:        /
+Activity.Kind:               Server
+Activity.StartTime:          2023-06-16T04:50:26.7672615Z
+Activity.Duration:           00:00:00.0121259
+Activity.Tags:
+    net.host.name: localhost
+    net.host.port: 7275
+    http.method: GET
+    http.scheme: https
+    http.target: /
+    http.url: https://localhost:7275/
+    http.flavor: 1.1
+    http.user_agent: curl/8.0.1
+    http.status_code: 200
+Resource associated with Activity:
+    service.name: OTel-Prometheus-Graphana-Yaeger
+    service.instance.id: e1afb619-bc32-48d8-b71f-ee196dc2a76a
+    telemetry.sdk.name: opentelemetry
+    telemetry.sdk.language: dotnet
+    telemetry.sdk.version: 1.5.0
+```
+
+The first is the inner / custom activity we created, the second is created by ASP.NET for the request and includes tags for the http request properties. You will see that both have the same `TraceId`, this identifies a single transaction and in a distributed system can be used to correlate the traces from each service. The id's are transmitted as http headers, and ASP.NET Core will assign one if none is present when it recieves a request. HttpClient will include the headers by default on outbound requests. The Greeter activity is parented to the other activity through the `ParentSpanId` which maps to the `SpanId` of the http activity.
+
+In a later stage we will feed this data into Jaeger to visualize the distributed traces.
+
+### 7. Install and use Prometheus
```

Comment:
Is this an opportunity to mention Azure Managed Prometheus?
https://learn.microsoft.com/azure/azure-monitor/essentials/prometheus-metrics-overview

Reply:
I am trying to strike a balance between showing the OSS options and our hosted solutions. We seem to usually swing too hard in one direction or the other - this is intended to be product docs rather than marketing material. I am hoping that the with the App Insights part being so much simpler, it does a good job of selling the service.